In [1]:
import numpy as np
import pandas as pd

In [2]:
lower_bond = [33,26,19,12,5]
upper_bond = [39,32,25,18,11]
frekuensi = [2,8,19,20,11] 

data_frekuensi = pd.DataFrame({'lower_bond':lower_bond, 'upper_bond':upper_bond, 'frekuensi':frekuensi})
data_frekuensi

lower_bond  upper_bond  frekuensi
0          33          39          2
1          26          32          8
2          19          25         19
3          12          18         20
4           5          11         11

## Setup midle value in class

In [3]:
data_frekuensi['midle_value'] = np.array((data_frekuensi.lower_bond + data_frekuensi.upper_bond)/2, dtype=np.int64)
data_frekuensi

lower_bond  upper_bond  frekuensi  midle_value
0          33          39          2           36
1          26          32          8           29
2          19          25         19           22
3          12          18         20           15
4           5          11         11            8


## Setup f . X

In [4]:
data_frekuensi['fX'] = data_frekuensi.frekuensi * data_frekuensi.midle_value
data_frekuensi

lower_bond  upper_bond  frekuensi  midle_value   fX
0          33          39          2           36   72
1          26          32          8           29  232
2          19          25         19           22  418
3          12          18         20           15  300
4           5          11         11            8   88

## Masukan ke rumus mean untuk data table frekuensi

In [5]:
average = sum(data_frekuensi.fX) / sum (frekuensi)
average

18.5

Kita mendapati nilai rata-rata pada hipmunan distribusi frekuensi di atas adalah **18.5**